src/Machine.js

```diff
@@ -0,0 +1,105 @@
+import { Console, Random } from "@woowacourse/mission-utils";
+import { moneyValidation, winNumbersValidation, bonusNumberValidation } from "./validation.js";
+import { USER_MESSAGE, LOTTO_PRICE, PERCENT_FACTOR, RESULT_COMMENTS, RESULT_MONEY } from "./constants.js";
+import Lotto from "./Lotto.js";
+
+class Machine {
+  #money;
+  #lottoList;
+  #winNumbers;
+  #bonusNumber;
+  #lottoResults;
+  #totalMoney;
+
+  constructor() {
+    this.#money = 0;
+    this.#lottoList = [];
+    this.#winNumbers = [];
+    this.#bonusNumber = 0;
+    this.#lottoResults = [
+      [RESULT_COMMENTS.FIRST, RESULT_MONEY.FIRST, 0],
+      [RESULT_COMMENTS.SECOND, RESULT_MONEY.SECOND, 0],
+      [RESULT_COMMENTS.THIRD, RESULT_MONEY.THIRD, 0],
+      [RESULT_COMMENTS.FOURTH, RESULT_MONEY.FOURTH, 0],
+      [RESULT_COMMENTS.FIFTH, RESULT_MONEY.FIFTH, 0],
+    ];
+    this.#totalMoney = 0;
+  }
+
+  async play() {
+    await this.readUserMoney();
+    await this.readUserWinNumbers();
+    await this.readUserBonusNumber();
+    this.lottoWinningCheck();
+    this.printBeforeResult();
+    this.calculatorMoney();
+    this.printProfit();
+  }
+
+  async readUserMoney() {
+    const moneyString =
+      await Console.readLineAsync(USER_MESSAGE.MONEY_INPUT);
+    this.#money = moneyValidation(moneyString);
+
+    this.#printBuyCount();
+  }
+
+  #printBuyCount() {
+    const buyCount = parseInt(this.#money, 10) / LOTTO_PRICE;
+    Console.print(`\n${buyCount}개를 구매했습니다.`);
+
+    this.#createLotto(buyCount);
+  }
+
+  #createLotto(buyCount) {
+    this.#lottoList = Array.from({ length: buyCount }, () => {
+      const randomNumbers = Random.pickUniqueNumbersInRange(1, 45, 6).sort(
+        (a, b) => a - b,
+      );
+
+      return new Lotto(randomNumbers);
+    });
+  }
+
+  async readUserWinNumbers() {
+    const winNumbersString = await Console.readLineAsync(USER_MESSAGE.WIN_NUMBERS_INPUT);
+    this.#winNumbers = winNumbersValidation(winNumbersString);
+  }
+
+  async readUserBonusNumber() {
+    const bonusNumberString = await Console.readLineAsync(USER_MESSAGE.BONUS_NUMBER_INPUT);
+    this.#bonusNumber = bonusNumberValidation(bonusNumberString);
+  }
+
+  lottoWinningCheck() {
+    this.#lottoList.forEach((lotto) => {
+      const rank = lotto.matchRank(this.#winNumbers, this.#bonusNumber);
+      if (rank) this.#lottoResults[rank - 1][2] += 1;
+    });
+  }
+
+  calculatorMoney() {
+    this.#totalMoney = this.#lottoResults.reverse().reduce((acc, cur) => {
```

Comment:
우선 reverse() 라는 메서드는 내장 메서드는 배열의 원본을 바꾸는 경향이 있는데요.
해당 로직이 이루어지고 나서 this.#lottoResults 의 값은 reverse 처리되어 배열이 역순으로 바껴있을거에요

그렇다면 나중에 this.#lottoResults 의 값을 사용할 때 예상치 못할수도 있어 최대한 값을 안바꾸는게 좋다고 생각해요

```suggestion
    this.#totalMoney = this.#lottoResults.slice().reverse().reduce((acc, cur) => {
```

이렇게 하면 배열을 `this.#lottoResults`의 원본을 보장할 수 있어요